pkg/git/gogit/checkout.go

```diff
@@ -153,10 +159,15 @@ func (c *CheckoutTag) Checkout(ctx context.Context, path, url string, opts *git.
 		}
 
 		if currentRevision != "" && currentRevision == c.LastRevision {
-			return nil, git.NoChangesError{
-				Message:          "no changes since last reconcilation",
-				ObservedRevision: currentRevision,
+			// Construct a partial commit with the existing information.
+			// Split the revision and take the last part as the hash.
+			// Example revision: 6.1.4/bf09377bfd5d3bcac1e895fa8ce52dc76695c060
+			ss := strings.Split(currentRevision, "/")
+			c := &git.Commit{
+				Hash:      git.Hash(ss[len(ss)-1]),
```

Comment:
Would there be any scenario in which `currentRevision` may not have the `/`, resulting in index out of bounds?

Reply:
That's created using https://github.com/fluxcd/source-controller/blob/e180b3c9446077559dfd3abf6e81cc8f8702df9f/pkg/git/git.go#L67-L72, less likely to happen, which is passed to `NewArtifactFor()` https://github.com/fluxcd/source-controller/blob/e180b3c9446077559dfd3abf6e81cc8f8702df9f/controllers/gitrepository_controller.go#L515 .